modules/sfp_talosintel.py

```diff
@@ -52,136 +42,116 @@ class sfp_talosintel(SpiderFootPlugin):
-
-                # If we're looking at hostnames/domains/IPs
-                if 'regex' not in malchecks[check]:
-                    for line in data['content'].split('\n'):
-                        if line == target or (targetType == "domain" and line == targetDom):
-                            self.sf.debug(target + "/" + targetDom + " found in " + check + " list.")
-                            return url
-                else:
-                    # Check for the domain and the hostname
-                    try:
-                        rxDom = str(malchecks[check]['regex']).format(targetDom)
-                        rxTgt = str(malchecks[check]['regex']).format(target)
-                        for line in data['content'].split('\n'):
-                            if (targetType == "domain" and re.match(rxDom, line, re.IGNORECASE)) or \
-                                    re.match(rxTgt, line, re.IGNORECASE):
-                                self.sf.debug(f"{target}/{targetDom} found in {check} list.")
-                                return url
-                    except Exception as e:
-                        self.sf.debug("Error encountered parsing 2: " + str(e))
-                        continue
-
-        return None
-
-    def lookupItem(self, resourceId, itemType, target):
-        for check in list(malchecks.keys()):
-            cid = malchecks[check]['id']
-            if cid == resourceId and itemType in malchecks[check]['checks']:
-                self.sf.debug(f"Checking maliciousness of {target} ({itemType}) with: {cid}")
-                return self.resourceList(cid, target, itemType)
-
-        return None
+        return [
+            "MALICIOUS_IPADDR",
+            "MALICIOUS_AFFILIATE_IPADDR",
+            "MALICIOUS_SUBNET",
+            "MALICIOUS_NETBLOCK"
+        ]
+
+    def queryBlacklist(self, target, targetType):
+        blacklist = self.retrieveBlacklist()
+
+        if targetType == "ip":
+            if target in blacklist:
+                self.sf.debug(f"IP address {target} found in Talos Intelligence blacklist.")
+                return True
+        elif targetType == "netblock":
+            netblock = IPNetwork(target)
+            for ip in blacklist:
+                if IPAddress(ip) in netblock:
+                    self.sf.debug(f"IP address {ip} found within netblock/subnet {target} in Talos Intelligence blacklist.")
+                    return True
+
+        return False
+
+    def retrieveBlacklist(self):
+        blacklist = self.sf.cacheGet('talosintel', 24)
+
+        if blacklist is not None:
+            return self.parseBlacklist(blacklist)
+
+        # https://talosintelligence.com/documents/ip-blacklist redirects to:
+        # https://snort.org/downloads/ip-block-list
+        res = self.sf.fetchUrl(
+            "https://snort.org/downloads/ip-block-list",
+            self.opts['_fetchtimeout'],
+            useragent=self.opts['_useragent'],
+        )
```

Comment:
`https://talosintelligence.com/documents/ip-blacklist` was giving me HTTP 503 errors. I'm not sure if it was intermittent or due to blocking the user agent or due to using HTTP/2.0. Using the direct URL works.